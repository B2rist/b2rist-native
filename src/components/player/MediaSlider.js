import React, {useMemo} from 'react';
import {Platform, StyleSheet} from 'react-native';
import {Text} from 'react-native-paper';
import Slider from '@react-native-community/slider';
import PropTypes from 'prop-types';
import {getAudioTimeString} from 'utils/timeUtils';

const MediaSlider = ({
  onSliderEditStart,
  onSliderEditEnd,
  onSliderEditing,
  playSeconds,
  duration,
}) => {
  const durationString = useMemo(
    () => getAudioTimeString(duration),
    [duration],
  );
  const currentTimeString = useMemo(
    () => getAudioTimeString(playSeconds),
    [playSeconds],
  );

  return (
    <>
      <Text style={styles.time}>{currentTimeString}</Text>
      <Slider
        onTouchStart={onSliderEditStart}
        onTouchEnd={onSliderEditEnd}
        onSlidingComplete={onSliderEditing}
        value={playSeconds}
        maximumValue={duration}
        maximumTrackTintColor="gray"
        minimumTrackTintColor="white"
        thumbTintColor="white"
        style={styles.slider}
      />
      <Text style={styles.duration}>{durationString}</Text>
    </>
  );
};

const styles = StyleSheet.create({
  time: {
    color: 'white',
    alignSelf: 'center',
  },
  slider: {
    flex: 1,
    alignSelf: 'center',
    marginHorizontal: Platform.select({ios: 5}),
  },
  duration: {
    color: 'white',
    alignSelf: 'center',
  },
});

export default React.memo(MediaSlider);

MediaSlider.propTypes = {
  onSliderEditStart: PropTypes.func.isRequired,
  onSliderEditEnd: PropTypes.func.isRequired,
  onSliderEditing: PropTypes.func.isRequired,
  playSeconds: PropTypes.number.isRequired,
  duration: PropTypes.number.isRequired,
};
