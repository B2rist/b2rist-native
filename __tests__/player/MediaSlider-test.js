import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MediaSlider from 'components/player/MediaSlider';

const DEFAULT_PROPS = {
  onSliderEditStart: jest.fn(),
  onSliderEditEnd: jest.fn(),
  onSliderEditing: jest.fn(),
  playSeconds: 1,
  duration: 150,
};

const createRenderer = props =>
  renderer.create(<MediaSlider {...props} />).toJSON();

describe('media slider tests', () => {
  it('media slider renders test', () => {
    const slider = createRenderer(DEFAULT_PROPS);
    expect(slider).toMatchSnapshot();
  });
});
