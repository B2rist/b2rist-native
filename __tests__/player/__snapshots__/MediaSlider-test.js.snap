// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media slider tests media slider renders test 1`] = `
[
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "alignSelf": "center",
          "color": "white",
        },
      ]
    }
  >
    00:00:01
  </Text>,
  <RNCSlider
    disabled={false}
    inverted={false}
    lowerLimit={-9007199254740991}
    maximumTrackTintColor="gray"
    maximumValue={150}
    minimumTrackTintColor="white"
    minimumValue={0}
    onChange={null}
    onRNCSliderAccessibilityAction={null}
    onRNCSliderSlidingComplete={[Function]}
    onRNCSliderSlidingStart={null}
    onRNCSliderValueChange={null}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTouchEnd={[MockFunction]}
    onTouchStart={[MockFunction]}
    step={0}
    style={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginHorizontal": 5,
        },
        {
          "height": 40,
        },
      ]
    }
    tapToSeek={false}
    thumbTintColor="white"
    upperLimit={9007199254740991}
    value={1}
  />,
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "alignSelf": "center",
          "color": "white",
        },
      ]
    }
  >
    00:02:30
  </Text>,
]
`;
